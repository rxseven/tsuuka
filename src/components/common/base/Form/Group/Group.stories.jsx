import React from 'react';
import { storiesOf } from '@storybook/react';

import Group from './index';

const title = 'Components/Common/Base/Form/Group';

storiesOf(title, module)
  .addParameters({
    info: {
      text: `
  **Usage:** for more information, see [Bootstrap - Groups](https://getbootstrap.com/docs/4.1/components/forms/).
`
    }
  })
  .addDecorator(story => <form>{story()}</form>)
  .add('default', () => (
    <Group>
      <label htmlFor="email">Username</label>
      <input
        className="form-control"
        id="email"
        placeholder="Your personal email address"
        type="text"
      />
    </Group>
  ));
