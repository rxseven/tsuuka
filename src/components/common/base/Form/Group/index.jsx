import cx from 'classnames';
import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  className: PropTypes.string
});

const defaultProps = {
  className: ''
};

function Group({ children, className }) {
  return <div className={cx('form-group', className)}>{children}</div>;
}

Group.propTypes = propTypes;
Group.defaultProps = defaultProps;

export default Group;
