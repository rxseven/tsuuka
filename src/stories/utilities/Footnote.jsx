import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
