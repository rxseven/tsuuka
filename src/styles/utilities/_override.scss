// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
