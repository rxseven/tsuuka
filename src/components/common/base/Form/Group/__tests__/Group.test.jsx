import React from 'react';

import { factory, toMarkup } from 'tests/utilities';
import Group from '../index';

// Arrange
const seed = { content: 'content' };
const source = { children: <span>{seed.content}</span> };
const input = { ...seed, ...source };

// Setup
function setup(props) {
  return factory(Group, source, props);
}

// Test suites
describe('<Form.Group />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed children correctly', () => {
    const expected = {
      content: input.content,
      html: input.children
    };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
    expect(component).toContainHTML(toMarkup(expected.html));
  });

  it('should render form with custom class names', () => {
    const props = { className: 'custom-class' };
    const expected = { class: props.className };
    const { component } = setup(props);

    expect(component).toHaveClass(expected.class);
  });
});
