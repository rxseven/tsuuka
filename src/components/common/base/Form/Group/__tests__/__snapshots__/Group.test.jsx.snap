// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form.Group /> Snapshot tests should render correctly 1`] = `
<div
  className="form-group"
>
  <div>
    Form controls
  </div>
</div>
`;
